Library/Homebrew/sandbox.rb

```diff
@@ -0,0 +1,95 @@
+require "erb"
+require "tempfile"
+
+class Sandbox
+  SANDBOX_EXEC = "/usr/bin/sandbox-exec".freeze
+
+  def self.available?
+    OS.mac? && File.executable?(SANDBOX_EXEC)
+  end
+
+  def initialize(formula=nil)
+    @profile = SandboxProfile.new
+    unless formula.nil?
+      allow_write "/private/tmp", :type => :subpath
+      allow_write "/private/var/folders", :type => :subpath
+      allow_write HOMEBREW_TEMP, :type => :subpath
+      allow_write HOMEBREW_LOGS/formula.name, :type => :subpath
+      allow_write HOMEBREW_CACHE, :type => :subpath
+      allow_write formula.rack, :type => :subpath
+      allow_write HOMEBREW_PREFIX/"etc", :type => :subpath
+      allow_write HOMEBREW_PREFIX/"var", :type => :subpath
```

Comment:
Actually, it'd be good if this used the `etc` value from `Formula` so that files written to `etc` can be correctly bottled: https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/formula.rb#L301

We perhaps also want similar logic for `var` in `Formula`.